Editor/Window/DeploymentScenariosInput.cs

```diff
@@ -44,6 +45,15 @@ public DeploymentScenariosInput(VisualElement container, DeploymentScenarios dep
                 _isExpanded = true;
                 UpdateGUI();
             });
+            stateManager.OnUserProfileUpdated += () =>
+            {
+                SetValue(stateManager.DeploymentScenario);
+                if (stateManager.DeploymentScenario != DeploymentScenarios.SingleRegion)
+                {
+                    _isExpanded = true;
+                }
+                UpdateGUI();
+            };
```

Comment:
This seems to be in reverse to me. The DeploymentScenario changing is controlled by `DeploymentScenariosInput`, so there should be no need to add this logic to the `OnUserProfileUpdated` callback which results in more processing. 

I would think that when the radio is changed, you ensure that the StateManager is also updated, and not the other way round which is currently the case.

Reply:
This change covers the journey that you change profile through the profile selector,. Since selected scenario is tied to profile, we need to get the scenario that is associated with that profile and set it.

Reply:
I'm moving lines 50-53 within UpdateGUI. Since this is logic to refresh the UI, it will be triggered whenever a UI refresh is expected, and hopefully avoid some bugs